### PR TITLE
[Usage stats] Add tags & number of nodes to the report.

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This PR adds the `RAY_EXTRA_USAGE_TAGS` to add additional tag metadata + number of nodes to the report.

NOTE: We shouldn't merge this until we update the backend validation. 

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — APPROVED** [review]

**Comment by rkooo567** [comment]
Please do not merge this now!

**Comment by rkooo567** [comment]
Hey @ericl to be clear, it is not required to pick for 1.13.1 right?

**Comment by ericl** [comment]
Yeah, I think this one is okay, since it's a new feature and not fixing a bug. Also, the implementation is complex enough that I think it's questionable for a patch release.

File: python/ray/_private/usage/usage_lib.py [hunk 1]
```diff
@@ -132,6 +132,12 @@ class UsageStatsToReport:
     total_failed: int
     #: The sequence number of the report.
     seq_number: int
+    #: The extra tags to report when specified by an
+    #  environment variable EXTRA_USAGE_TAGS
+    extra_usage_tags: Dict[str, str]
+    #: The number of alive nodes when the report is generated.
+    # -1 means it fails to get the number of nodes in the cluster.
+    total_num_nodes: int
```
Comment (jjyao) [inline python/ray/_private/usage/usage_lib.py:140]: Optional[int]

File: python/ray/_private/usage/usage_lib.py [hunk 2]
```diff
@@ -132,6 +132,12 @@ class UsageStatsToReport:
     total_failed: int
     #: The sequence number of the report.
     seq_number: int
+    #: The extra tags to report when specified by an
+    #  environment variable EXTRA_USAGE_TAGS
+    extra_usage_tags: Dict[str, str]
```
Comment (jjyao) [inline python/ray/_private/usage/usage_lib.py:137]: Optional[Dict[str, str]]

File: python/ray/_private/usage/usage_lib.py [hunk 3]
```diff
@@ -132,6 +132,12 @@ class UsageStatsToReport:
...
+    #: The number of alive nodes when the report is generated.
+    # -1 means it fails to get the number of nodes in the cluster.
```
Comment (jjyao) [inline python/ray/_private/usage/usage_lib.py:139]: We use None to indicate missing value.

**Review by jjyao — APPROVED** [review]

**Pull request merged**